src/Caching/SqlServer/src/DatabaseOperations.cs

```diff
@@ -259,25 +281,80 @@ public async Task SetCacheItemAsync(string key, byte[] value, DistributedCacheEn
 
             await connection.OpenAsync(token).ConfigureAwait(false);
 
-            using (var reader = await command.ExecuteReaderAsync(
-                CommandBehavior.SequentialAccess | CommandBehavior.SingleRow | CommandBehavior.SingleResult,
-                token).ConfigureAwait(false))
+            if (includeValue)
             {
+                using var reader = await command.ExecuteReaderAsync(
+                    CommandBehavior.SequentialAccess | CommandBehavior.SingleRow | CommandBehavior.SingleResult, token).ConfigureAwait(false);
+
                 if (await reader.ReadAsync(token).ConfigureAwait(false))
                 {
-                    if (includeValue)
+                    if (destination is null)
                     {
                         value = await reader.GetFieldValueAsync<byte[]>(Columns.Indexes.CacheItemValueIndex, token).ConfigureAwait(false);
                     }
+                    else
+                    {
+                        StreamOut(reader, Columns.Indexes.CacheItemValueIndex, destination);
+                        value = []; // use non-null here as a sentinel to say "we got one"
```

Comment:
I don't know what `value` represents, but I'm assuming `[]` is not a valid value for it.

Reply:
In the non-streaming API, the result is either a right-sized array or null. With the streaming API we don't need to return he contents in an array, but we still need to signal "found" Vs "not found", so we'll use "returned a non-null array" for that, to avoid complicating the API.